app/src/main/java/org/p2p/wallet/newsend/ui/search/NewSearchPresenter.kt

```diff
@@ -186,20 +173,11 @@ class NewSearchPresenter(
             return
         }
 
-        val newAddresses = searchInteractor.searchByAddress(
-            publicKey.toBase58().toBase58Instance(),
-            initialToken
-        )
-
-        state.updateSearchResult(address, newAddresses)
+        val newAddresses = searchInteractor.searchByAddress(publicKey.toBase58(), initialToken)
```

Comment:
toBase58Instance?

Reply:
the purpose of Base58String is to remove overuse of primiites and use **types** instead of some kind unknown what's inside godly String

Reply:
Imho `Base58String` is super cool for cases with public keys, when you a lot of parameters and you may accidentally pass a wrong address. 

But in some cases you have a function which needs only one address and uses it in different functions inside.

Reply:
reverted back right now, let's talk later